snafu/dns_perf_wrapper/trigger_dns_perf.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import time
+import logging
+import subprocess
+from ttp import ttp
+
+logger = logging.getLogger("snafu")
+
+
+class Trigger_dns_perf:
+    def __init__(self, args):
+        """ Initialize the arguments defined """
+
+        self.uuid = args.uuid
+        self.cluster_name = args.cluster_name
+        self.server_address = args.server_address
+        self.queries_per_second = args.queries_per_second
+        self.run_time = args.run_time
+        self.data_file = args.data_file
+        self.clients = args.clients
+
+    def _json_payload(self, data, timestamp, elapsed_time):
+        """ Generates a json payload with the metrics of interest.
+        This is the results document, it gets updated with metrics from the dns-perf run.
+        """
+
+        payload = {
+            "uuid": self.uuid,
+            "cluster_name": self.cluster_name,
+            "clients": self.clients,
+            "timestamp": timestamp,
+            "elapsed_time": elapsed_time,
+        }
+        try:
+            payload.update(data)
+        except Exception as e:
+            logging.error("Failed to update the result document, error: %s" % (e))
+        return payload
+
+    def run(self):
+        """ Runs dns-perf workload and returns the stdout """
+
+        command = "dnsperf -l {} -s {} -Q {} -d {} -c {}".format(
+            self.run_time, self.server_address, self.queries_per_second, self.data_file, self.clients
+        )
+        logger.info(command)
+        out = subprocess.Popen(
+            command, shell=True, universal_newlines=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
```

Comment:
Just curious, is there a reason you're using shell here? It has security implications.